fdbclient/NativeAPI.actor.cpp

```diff
@@ -3124,38 +3129,39 @@ ACTOR Future<Void> warmRange_impl(Reference<TransactionState> trState, KeyRange
 	return Void();
 }
 
-SpanID generateSpanID(bool transactionTracingSample, SpanID parentContext = SpanID()) {
-	uint64_t txnId = deterministicRandom()->randomUInt64();
+// TODO - ljoswiak and mpilman. This will need some close inspection.
+SpanContext generateSpanID(bool transactionTracingSample, SpanContext parentContext = SpanContext()) {
+	UID txnId = UID(deterministicRandom()->randomUInt64(), deterministicRandom()->randomUInt64());
 	if (parentContext.isValid()) {
-		if (parentContext.first() > 0) {
-			txnId = parentContext.first();
+		if (parentContext.traceID.first() > 0) {
+			txnId = parentContext.traceID;
 		}
-		uint64_t tokenId = parentContext.second() > 0 ? deterministicRandom()->randomUInt64() : 0;
-		return SpanID(txnId, tokenId);
+		uint64_t spanId = parentContext.spanID > 0 ? deterministicRandom()->randomUInt64() : 0;
+		return SpanContext(txnId, spanId);
 	} else if (transactionTracingSample) {
-		uint64_t tokenId = deterministicRandom()->random01() <= FLOW_KNOBS->TRACING_SAMPLE_RATE
-		                       ? deterministicRandom()->randomUInt64()
-		                       : 0;
-		return SpanID(txnId, tokenId);
+		uint64_t spanId = deterministicRandom()->random01() <= FLOW_KNOBS->TRACING_SAMPLE_RATE
+		                      ? deterministicRandom()->randomUInt64()
+		                      : 0;
+		return SpanContext(txnId, spanId, spanId > 0 ? TraceFlags::sampled : TraceFlags::unsampled);
```

Comment:
I don't think this matters as much, but `spanId` should probably always be a randomly generated value here, and then the sample flag is set to true/false depending on the condition `deterministicRandom()->random01() <= FLOW_KNOBS->TRACING_SAMPLE_RATE` (I say it doesn't matter much because it probably won't matter if the span ID is 0 if no sample is being taken).